src/components/Search/Search.tsx

```diff
@@ -0,0 +1,33 @@
+import React from 'react';
+import { Icon, Input } from 'semantic-ui-react';
+import { debounce } from '../../helpers';
+
+type PropsSearch = {
+  setSearchQuery: (query: string) => void;
+};
+
+export const Search: React.FC<PropsSearch> = ({ setSearchQuery }) => {
+  const debouncedSearch = debounce(
+    (value: string) => setSearchQuery(value), 1000,
+  );
+
+  const handleInputChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const value = event.target.value.toLowerCase().trim().slice(0, 20);
```

Comment:
why do you slice the string?

Reply:
So that the search query is not more than 20 characters, this is my application implementation